libafl_frida/src/alloc.rs

```diff
@@ -98,6 +99,30 @@ impl Allocator {
         // probe to find a usable shadow bit:
         let mut shadow_bit = 0;
 
+        let mut occupied_ranges: Vec<(usize, usize)> = vec![];
+        // max(userspace address) this is usually 0x8_0000_0000_0000 - 1 on x64 linux.
+        let mut userspace_max: usize = 0;
+        // List up all occupied memory ranges
+        RangeDetails::enumerate_with_prot(PageProtection::Read, &mut |details| {
```

Comment:
Can we assume that all relevant regions are READable?

Reply:
no... 😄 
I changed it to enumerate writable & executable regions

Reply:
surely there is a better way than copy-pasting all that code?

Reply:
do you mean using `PageProtection::ReadWriteExecute` instead?
but that does not work.
I need to enumerate Readable ,Writable, and Executable one by one and put them into a single `Vec`

Reply:
So maybe do a nested for instead of copying the code three times?

Reply:
ah ok